backend/src/test/java/de/neuefische/backend/service/RestCountriesServiceTest.java

```diff
@@ -0,0 +1,78 @@
+import de.neuefische.backend.exception.RestCountryException;
+import de.neuefische.backend.model.Country;
+import de.neuefische.backend.model.restcountries.RestCountry;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class RestCountriesServiceTest {
+
+    RestCountriesService restCountriesService = new RestCountriesService();
+
+    @Test
+    void refactorApiResponse_deseralize_properly() throws JsonProcessingException, RestCountryException {
+        String givenJson = """
+                [
+                    {
+                        "name": {
+                            "common": "Mauritania",
+                            "official": "Islamic Republic of Mauritania",
+                            "nativeName": {
+                                "ara": {
+                                    "official": "الجمهورية الإسلامية الموريتانية",
+                                    "common": "موريتانيا"
+                                }
+                            }
+                        },
+                        "cca3": "MRT",
+                        "capital": [
+                            "Nouakchott"
+                        ],
+                        "altSpellings": [
+                            "MR",
+                            "Islamic Republic of Mauritania",
+                            "al-Jumhūriyyah al-ʾIslāmiyyah al-Mūrītāniyyah"
+                        ],
+                        "flag": "🇲🇷"
+                    },
+                    {
+                        "name": {
+                            "common": "Aruba",
+                            "official": "Aruba",
+                            "nativeName": {
+                                "nld": {
+                                    "official": "Aruba",
+                                    "common": "Aruba"
+                                },
+                                "pap": {
+                                    "official": "Aruba",
+                                    "common": "Aruba"
+                                }
+                            }
+                        },
+                        "cca3": "ABW",
+                        "capital": [
+                            "Oranjestad"
+                        ],
+                        "altSpellings": [
+                            "AW"
+                        ],
+                        "flag": "🇦🇼"
+                    }
+                    ]
+                """;
+        ObjectMapper objectMapper = new ObjectMapper();
+        RestCountry[] restCountries = objectMapper.readValue(givenJson, RestCountry[].class);
+
+        List<Country> countries = restCountriesService.refactorApiResponse(restCountries);
+        assertThat(countries.get(0).name()).isEqualTo("Mauritania");
+        assertThat(countries.get(0).flag()).isEqualTo("🇲🇷");
+        assertThat(countries.get(0).threeLetterCode()).isEqualTo("MRT");
```

Comment:
Why do you only check for Mauritania? And not for the whole list? There is a second element aswell

Reply:
when the first is parsed properly, the second one should be too.

Reply:
Yeah you should be right, but i would make sure im right. And its not that much work